File: apps/warehouse/services/warehouse.py
from ..models.warehouse import Warehouse

class WarehouseService:
    """
    Business logic layer for Warehouse.
    Decouples database operations from views.
    """
    @staticmethod
    def get_all():
        return Warehouse.objects.all()

    @staticmethod
    def get_by_id(obj_id):
        return Warehouse.objects.filter(id=obj_id).first()

    @staticmethod
    def create(data):
        return Warehouse.objects.create(**data)

    @staticmethod
    def update(obj_id, data):
        obj = Warehouse.objects.filter(id=obj_id).first()
        if obj:
            for key, value in data.items():
                setattr(obj, key, value)
            obj.save()
        return obj

    @staticmethod
    def delete(obj_id):
        obj = Warehouse.objects.filter(id=obj_id).first()
        if obj:
            obj.delete()
            return True
        return False

File: apps/warehouse/frontend/src/services/warehouse.js
import http from './http'

export function getAll() {

  return http.get('/api/warehouses/')
}

export function getById(id) {

  return http.get(`/api/warehouses/${id}/`)
}

export function create(data) {

  return http.post('/api/warehouses/', data)
}

export function update(id, data) {

  return http.put(
    `/api/warehouses/${id}/`,
    data
  )
}

export function remove(id) {

  return http.delete(
    `/api/warehouses/${id}/`
  )
}